iset.mm

```diff
@@ -127782,6 +127864,740 @@ reduced fraction representation (no common factors, denominator
+        ZUQUFZSZOTURZUATSZKWOWTNWLSZOTURZGTSXEWNXGGTWNWGWSUPZNWLSZFTURXGWMXIFTW
+        JXHENWLENRWIWSWGWHWRHUSUTVBVAXIXFFOTFORZXHWTNWLXJWGWQWSWFWPHUSVCVDVEVFV
+        GXGXDGUATGUARZXFXCOTXKWTNWLXBWKXAQUQVHVIVJVBVFUNVSLUCLPZUDUEUFZUGZVLVTX
+        NQUJVLUBLUCVRXMVQXLUDUEVKVMQVSXNVNVOVP $.
+    $}
+  $}
+
+  ${
+    $d A f g j n $.  $d A g n x y $.  $d f g j k n $.  $d j n y $.
+    $( Lemma for ~ ennnfone .  The trivial direction.  (Contributed by Jim
+       Kingdon, 27-Oct-2022.) $)
+    ennnfonelemim $p |- ( A ~~ NN -> ( A. x e. A A. y e. A DECID x = y
+      /\ E. f ( f : NN0 -onto-> A
+        /\ A. n e. NN0 E. k e. NN0 A. j e. ( 0 ... n )
+        ( f ` k ) =/= ( f ` j ) ) ) ) $=
+      ( vg cn wbr cn0 cv wceq wral cfv wa wcel syl2anc wi adantl cen wdc wfo co
+      wne cc0 cfz wrex wex nn0ennn ensymi entr mpan2 wf1o bren biimpi cz adantr
+      wf f1of simprl ffvelrnd nn0zd simprr zdceq wfn crn dff1o6 r19.21bi anasss
+      simp3bi fveq2 impbid1 dcbid mpbid ralrimivva ccnv f1ocnv syl c1 peano2nn0
+      f1ofo caddc elfznn0 nn0red cle clt elfzle2 wb simplr nn0leltp1 neneqd w3a
+      gtned sylib simp3d ad2antrr fveqeq2 eqeq1 imbi12d eqeq2d eqeq2 rspc2v mpd
+      neqned ralrimiva neeq1d ralbidv rspcev cvv cnvexg elv foeq1 fveq1 neeq12d
+      mtod rexralbidv anbi12d spcev jca exlimddv ) CIUAJZCKUAJZALZBLZMZUBZBCNAC
+      NZKCDLZUCZFLZYIOZELZYIOZUEZEUFGLZUGUDZNFKUHZGKNZPZDUIZPZYBIKUAJYCKIUJUKCI
+      KULUMYCCKHLZUNZUUBHYCUUDHUICKHUOUPUUDUUBYCUUDYHUUAUUDYGABCCUUDYDCQZYECQZP
+      ZPZYDUUCOZYEUUCOZMZUBZYGUUHUUIUQQUUJUQQUULUUHUUIUUHCKYDUUCUUDCKUUCUSUUGCK
+      UUCUTURZUUDUUEUUFVAVBVCUUHUUJUUHCKYEUUCUUMUUDUUEUUFVDVBVCUUIUUJVERUUHUUKY
+      FUUHUUKYFUUDUUEUUFUUKYFSZUUDUUEPUUNBCUUDUUNBCNZACUUDUUCCVFUUCVGKMUUOACNAB
+      CKUUCVHVKVIVIVJYDYEUUCVLVMVNVOVPUUDKCUUCVQZUCZYKUUPOZYMUUPOZUEZEYQNZFKUHZ
+      GKNZUUAUUDKCUUPUNZUUQCKUUCVRZKCUUPWBVSUUDUVBGKUUDYPKQZPZYPVTWCUDZKQZUVHUU
+      POZUUSUEZEYQNZUVBUVFUVIUUDYPWATZUVGUVKEYQUVGYMYQQZPZUVJUUSUVOUVJUUSMZUVHY
+      MMZUVOUVHYMUVOYMUVHUVOYMUVNYMKQZUVGYMYPWDTZWEUVOYMYPWFJZYMUVHWGJZUVNUVTUV
+      GYMUFYPWHTUVOUVRUVFUVTUWAWIUVSUUDUVFUVNWJYMYPWKRVOWNWLUVOYDUUPOYEUUPOZMZY
+      FSZBKNAKNZUVPUVQSZUUDUWEUVFUVNUUDUUPKVFZUUPVGCMZUWEUUDUVDUWGUWHUWEWMUVEAB
+      KCUUPVHWOWPWQUVOUVIUVRUWEUWFSUVGUVIUVNUVMURUVSUWDUWFUVJUWBMZUVHYEMZSABUVH
+      YMKKYDUVHMUWCUWIYFUWJYDUVHUWBUUPWRYDUVHYEWSWTYEYMMZUWIUVPUWJUVQUWKUWBUUSU
+      VJYEYMUUPVLXAYEYMUVHXBWTXCRXDXPXEXFUVAUVLFUVHKYKUVHMZUUTUVKEYQUWLUURUVJUU
+      SYKUVHUUPVLXGXHXIRXFYTUUQUVCPDUUPUUPXJQHUUCXJXKXLYIUUPMZYJUUQYSUVCKCYIUUP
+      XMUWMYRUVBGKUWMYOUUTFEKYQUWMYLUURYNUUSYKYIUUPXNYMYIUUPXNXOXQXHXRXSRXTTYAV
+      S $.
+  $}
+
+  ${
+    $d A f j n x y $.  $d f j k n $.
+    $( A condition for a set being countably infinite.  Corollary 8.1.13 of
+       [AczelRathjen], p. 73.  (Contributed by Jim Kingdon, 27-Oct-2022.) $)
+    ennnfone $p |- ( A ~~ NN <-> ( A. x e. A A. y e. A DECID x = y
```

Comment:
Is there a short and concise description of the condition? What is the different/advantage compared with f being a bijection (see ~bren)? Why is the condition about NN0, but the LHS contains NN?

Reply:
> Is there a short and concise description of the condition?

I've pushed up a stab at it. "A is countable and infinite" gets at the intuition but I can't imagine those exact words would ever get by you and @benjub (and not even me without some kind of qualification, which is why I expanded significantly on that thought in the comment I just tried to write).

> What is the different/advantage compared with f being a bijection (see ~bren)?

The best way to answer that will be to start using this theorem. I was looking at the way that AczelRathjen does this to prove `ZZ ~~ NN` and `QQ ~~ NN` and to be honest there were some steps in their proofs of those two things which seemed to me to be missing (or at least, not spelled out).

I might give a try at https://us.metamath.org/mpeuni/unben.html - that looks easier than `QQ ~~ NN`, and at least at first glance would appear to be a case where the theorem in this pull request might be easier than building the bijection directly (the set.mm proof of unben uses excluded middle - via https://us.metamath.org/mpeuni/unbnn.html ).

> Why is the condition about NN0, but the LHS contains NN?

LOL, I thought you'd ask about ` _om ` versus ` NN0 ` rather than ` NN0 ` versus ` NN ` .

Using ` NN ` in `A ~~ NN` is our convention in theorems like https://us.metamath.org/mpeuni/prminf.html , https://us.metamath.org/mpeuni/qnnen.html , and https://us.metamath.org/mpeuni/unben.html .

On the right side, I sort of think the best style might be ` _om ` as that is what we use in theorems like https://us.metamath.org/ileuni/finct.html and https://us.metamath.org/ileuni/df-omni.html . But some level of conversion between ` NN0 ` and ` _om ` is inevitable in theorems like this (`QQ` is expressed in terms of `ZZ` rather than `_om`, `seq` is expressed in terms of `ZZ`, etc). To be honest I stated `ennnfone` this way while we were still having a bunch of discussions about which one to use in some of these theorems and never really thought to revisit because I was very focused on how to prove it (a topic which I got stuck on multiple times before eventually finishing the proof). I guess I don't see any big need to change anything now, and indeed I sort of suspect time may produce greater clarity on what the most convenient way(s) to express it is.

Reply:
Thanks a lot for these explanations, its much clearer to me now (although I am very curious about the application of this theorem, as you indicated).

Reply:
Yeah, I'm curious too. It would be a whole lot easier if we could prove something like
```
( E. f f : _om -onto-> A ) /\ _om ~<_ A )
  -> A ~~ NN
```
but my (relatively cursory) attempts at proving that as a corollary of `ennnfone` didn't seem to go anywhere.

So I may take a break from this topic unless I get an idea or decide to dive back in looking for one.